test/OpenFeature.Tests/OpenFeatureClientTests.cs

```diff
@@ -540,5 +540,101 @@ public void ToFlagEvaluationDetails_Should_Convert_All_Properties()
+            client.Track(trackingEventName, trackingEventDetails: trackingEventDetails);
+
+            Assert.Equal(4, provider.GetTrackingInvocations().Count);
+            Assert.Equal(trackingEventName, provider.GetTrackingInvocations()[0].Item1);
+            Assert.Equal(trackingEventName, provider.GetTrackingInvocations()[1].Item1);
+            Assert.Equal(trackingEventName, provider.GetTrackingInvocations()[2].Item1);
+            Assert.Equal(trackingEventName, provider.GetTrackingInvocations()[3].Item1);
+
+            Assert.NotNull(provider.GetTrackingInvocations()[0].Item2);
+            Assert.NotNull(provider.GetTrackingInvocations()[1].Item2);
+            Assert.NotNull(provider.GetTrackingInvocations()[2].Item2);
+            Assert.NotNull(provider.GetTrackingInvocations()[3].Item2);
+
+            Assert.Equal(EvaluationContext.Empty.Count, provider.GetTrackingInvocations()[0].Item2!.Count);
+            Assert.Equal(EvaluationContext.Empty.Count, provider.GetTrackingInvocations()[1].Item2!.Count);
+            Assert.Equal(EvaluationContext.Empty.Count, provider.GetTrackingInvocations()[2].Item2!.Count);
+            Assert.Equal(EvaluationContext.Empty.Count, provider.GetTrackingInvocations()[3].Item2!.Count);
+
+            Assert.Equal(EvaluationContext.Empty.TargetingKey, provider.GetTrackingInvocations()[0].Item2!.TargetingKey);
+            Assert.Equal(EvaluationContext.Empty.TargetingKey, provider.GetTrackingInvocations()[1].Item2!.TargetingKey);
+            Assert.Equal(EvaluationContext.Empty.TargetingKey, provider.GetTrackingInvocations()[2].Item2!.TargetingKey);
+            Assert.Equal(EvaluationContext.Empty.TargetingKey, provider.GetTrackingInvocations()[3].Item2!.TargetingKey);
+
+            Assert.Null(provider.GetTrackingInvocations()[0].Item3);
+            Assert.Null(provider.GetTrackingInvocations()[1].Item3);
+            Assert.Equal(trackingEventDetails, provider.GetTrackingInvocations()[2].Item3);
+            Assert.Equal(trackingEventDetails, provider.GetTrackingInvocations()[3].Item3);
+        }
+
+        public static TheoryData<string, EvaluationContext?, EvaluationContext?, EvaluationContext?, string> GenerateMergeEvaluationContextTestData()
+        {
+            const string key = "key";
+            const string global = "global";
+            const string client = "client";
+            const string invocation = "invocation";
+            var globalEvaluationContext = new[] { new EvaluationContextBuilder().Set(key, "global").Build(), null };
+            var clientEvaluationContext = new[] { new EvaluationContextBuilder().Set(key, "client").Build(), null };
+            var invocationEvaluationContext = new[] { new EvaluationContextBuilder().Set(key, "invocation").Build(), null };
+
+            var data = new TheoryData<string, EvaluationContext?, EvaluationContext?, EvaluationContext?, string>();

```

Comment:
Took me a minute to wrap my head around this but it makes sense! :sweat_smile:

Reply:
Should I add some documentation to it?